backend/api/parse_pdf.py

```
@@ -127,152 +202,202 @@ async def extract_text_from_file(file: UploadFile) -> str:
-                    if e.response.status_code == 429:  # Rate limited
-                        await asyncio.sleep(5)  # Wait longer for rate limits
-                        raise e  # Let retry decorator handle it
-                    raise HTTPException(status_code=e.response.status_code, detail=f"LlamaParse API error: {e}")
+            finally:
+                # Clean up temporary file
+                try:
+                    os.unlink(temp_file_path)
+                except Exception as e:
+                    debug_log(f"[LLAMAPARSE] Warning: Could not delete temp file {temp_file_path}: {e}")
                     
         except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)